FirebaseTwitterAuthUI/FIRTwitterAuthUI.m

```diff
@@ -0,0 +1,143 @@
+ @return Resource bundle for the auth provider.
+ */
++ (NSBundle *)frameworkBundle {
+  static NSBundle *frameworkBundle = nil;
+  static dispatch_once_t predicate;
+  dispatch_once(&predicate, ^{
+    NSString *mainBundlePath = [[NSBundle mainBundle] resourcePath];
+    NSString *frameworkBundlePath =
+    [mainBundlePath stringByAppendingPathComponent:kBundleFileName];
+    frameworkBundle = [NSBundle bundleWithPath:frameworkBundlePath];
+    if (!frameworkBundle) {
+      frameworkBundle = [NSBundle mainBundle];
+    }
+  });
+  return frameworkBundle;
+}
+
+/** @fn imageNamed:
+ @brief Returns an image from the resource bundle given a resource name.
+ @param name The name of the image file.
+ @return The image object for the named file.
+ */
++ (UIImage *)imageNamed:(NSString *)name {
+  NSString *path = [[[self class] frameworkBundle] pathForResource:name ofType:@"png"];
+  return [UIImage imageWithContentsOfFile:path];
+}
+
+/** @fn localizedStringForKey:
+ @brief Returns the localized text associated with a given string key. Will default to english
+ text if the string is not available for the current localization.
+ @param key A string key which identifies localized text in the .strings files.
+ @return Localized value of the string identified by the key.
+ */
++ (NSString *)localizedStringForKey:(NSString *)key {
+  NSBundle *frameworkBundle = [[self class] frameworkBundle];
+  return [frameworkBundle localizedStringForKey:key value:nil table:kTableName];
+}
+
+#pragma mark - FIRAuthProviderUI
+
+- (NSString *)providerID {
+  return FIRTwitterAuthProviderID;
+}
+
+- (NSString *)shortName {
+  return @"Twitter";
+}
+
+- (NSString *)signInLabel {
+  return [[self class] localizedStringForKey:kSignInWithTwitter];
+}
+
+- (UIImage *)icon {
+  return [[self class] imageNamed:@"ic_twitter"];
+}
+
+- (UIColor *)buttonBackgroundColor {
+  return [UIColor colorWithRed:71.0f/255.0f green:154.0f/255.0f blue:234.0f/255.0f alpha:1.0f];
+}
+
+- (UIColor *)buttonTextColor {
+  return [UIColor whiteColor];
+}
+
+- (void)signInWithAuth:(FIRAuth *)auth
+                 email:(nullable NSString *)email
+presentingViewController:(nullable UIViewController *)presentingViewController
+            completion:(nullable FIRAuthProviderSignInCompletionBlock)completion {
+
+  [[Twitter sharedInstance]
+   logInWithCompletion:^(TWTRSession * _Nullable session, NSError * _Nullable error) {
+     if (session) {
+       FIRAuthCredential *credential =
+       [FIRTwitterAuthProvider credentialWithToken:session.authToken
+                                            secret:session.authTokenSecret];
+       if (completion) {
+         completion(credential, error);
+       }
+     } else {
+       NSError *newError =
+       [FIRAuthUIErrorUtils providerErrorWithUnderlyingError:error
+                                                  providerID:FIRTwitterAuthProviderID];
+       if (completion) {
+         completion(nil, error);
```

Comment:
fixed in both cases:
`     if (session) {
       FIRAuthCredential *credential =
       [FIRTwitterAuthProvider credentialWithToken:session.authToken
                                            secret:session.authTokenSecret];
       if (completion) {
         completion(credential, nil);
       }
     } else {
       NSError *newError =
       [FIRAuthUIErrorUtils providerErrorWithUnderlyingError:error
                                                  providerID:FIRTwitterAuthProviderID];
       if (completion) {
         completion(nil, newError);
       }
     }
`